geode-core/build.gradle

```diff
@@ -163,93 +165,159 @@ dependencies {
   jcaAnnotationProcessor(platform(project(':boms:geode-all-bom')))
   jmhCompile(platform(project(':boms:geode-all-bom')))
 
+  
+  //A dependency that contains the compiled output of the source. What is this for?
   classesOutput sourceSets.main.output
 
 
   // Source Dependencies
+  //------------------------------------------------------------
+
+  //  The antlr configuration is used by the antlr plugin, which compiles grammar
+  // files used by the query engine
   antlr 'antlr:antlr'
 
   // External
+  //------------------------------------------------------------
+
+
+  //tools.jar seems to be used by gfsh is some cases to control processes using
+  //the sun attach API? But this code path may not even be used?
   compileOnly files("${System.getProperty('java.home')}/../lib/tools.jar")
-  compile('com.github.stephenc.findbugs:findbugs-annotations')
-  compile('org.jgroups:jgroups')
-  compile('antlr:antlr')
-  compile('com.fasterxml.jackson.core:jackson-annotations')
-  compile('com.fasterxml.jackson.core:jackson-databind')
-  compile('commons-io:commons-io')
-  compile('commons-validator:commons-validator')
-  compile('commons-digester:commons-digester')
-  compile('com.sun.activation:javax.activation')
-  compile('javax.xml.bind:jaxb-api')
-  compile('com.sun.xml.bind:jaxb-impl')
-  compile('com.sun.istack:istack-commons-runtime') {
+
+  //Find bugs is used in multiple places in the code to suppress findbugs warnings
+  implementation('com.github.stephenc.findbugs:findbugs-annotations')
+
+  //Jgroups is a core component of our membership system.
+  implementation('org.jgroups:jgroups')
+
+  //Antlr is used by the query engine.
+  implementation('antlr:antlr')
+
+  //Jackson annotations is used in gfsh
+  implementation('com.fasterxml.jackson.core:jackson-annotations')
```

Comment:
I am surprised that none of the Jackson annotations appear in our public-facing classes. But in a good way